src/NodaTime.Test/PeriodTest.cs

```diff
@@ -931,6 +940,39 @@ public void Comparer_NonDurationablePeriods()
             Assert.AreEqual(0, februaryComparer.Compare(month, month));
         }
 
+        public static TestCaseData[] PeriodMaxAndMinValues
+        {
+            get
+            {
+                return [
+                    new TestCaseData(Period.MaxValue, int.MaxValue, long.MaxValue).SetName("All members should be MaxValue"), 
+                    new TestCaseData(Period.MinValue, int.MinValue, long.MinValue).SetName("All members should be MinValue")
+                ];
+            }
+        }
+
+        /// <summary>
+        /// Ensure that Period.MaxValue and Period.MinValue contain the max/min value assignable to each property/member.
+        /// </summary>
+        [Test]
+        [TestCaseSource(nameof(PeriodMaxAndMinValues))]
+        public void Period_MaxAndMinValues_AllMembers(Period period, int expectedIntValue, long expectedLongValue)
```

Comment:
I was thinking I'll work with reflection here but it just feels wrong to me.
This is the first time I'm working with NUnit so if there's changes needed here, please let me know.

Reply:
The reason for using reflection would be to avoid baking in assumptions about the type. If `Days` were actually a `long` rather than an `int`, the test would still pass. So I was planning on using all the properties, and testing the actual value against the expected value for that property's type.

I honestly don't know whether that's overkill - having explained my reasoning a bit more, what do you think?

The code would be something like this (which may not compile quite - I'm just typing it directly):

```csharp
foreach (var property in typeof(Period).GetProperties())
{
    var actualValue= property.GetValue(period);
    var expectedValue = actualValue is int ? expectedIntValue
        : actualValue is long ? expectedLongValue
        : throw new InvalidOperationException($"Property {property.Name} has unexpected type");
    Assert.AreEqual(expectedValue, actualValue);
}
```

Reply:
> The reason for using reflection would be to avoid baking in assumptions about the type

This is exactly what I was thinking and would try to do with every language. C# is a bit new to me, so I steer clear of reflection runtime sorcery.

Let me know if the new test works better.